app/controllers/budgets_controller.rb

```diff
@@ -1,6 +1,7 @@
 class BudgetsController < ApplicationController
+  before_action :set_grant
```

Comment:
I've tended to keep my controllers on the wetter side here to make code more readable and avoid additional before_actions in the controller.  This does mean a the same call to the model is required for all actions that require a grant to be set making code wet.  But the one call would all be in the action making the action easier to read, IMO.  This is my personal preference, what do you guys think?  At a min we should explicitly define before action only to apply to actions that require the grant to be set.  Doesn't make sense for "new" action, etc.